app/controllers/CheckDetailsController.scala

```diff
@@ -109,39 +111,82 @@ class CheckDetailsController @Inject()(
+  //    } else {
+  //      checkDetailsMatchingFailedWithUnknownIssue(mode)
+  //    }
+  //  }
+
   private def handleCheckDetailsFailureJourney(idDataError: IndividualDetailsError,
                                                mode: Mode,
                                                sessionWithNINO: Session,
                                                origin: Option[String],
-                                               pdvData: PDVResponseData)(implicit headerCarrier: HeaderCarrier): Result = {
-    if (pdvData.validationStatus.equals("")) {
-      Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
-    } else if (pdvData.validationStatus.equals("failure")) {
-      logger.warn(s"PDV matched failed: ${pdvData.validationStatus}")
-      auditService.findYourNinoPDVMatchFailed(pdvData, origin)
-      Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
-    } else {
-      val errorStatusCode: Option[String] = idDataError match {
-        case conError: ConnectorError => Some(conError.statusCode.toString)
-        case _ => None
-      }
+                                               pdvData: PDVResponseData
+                                              )(implicit hc: HeaderCarrier, request: DataRequest[AnyContent]): Result =
+    pdvData.validationStatus match {
+      case "failure" =>
+        logger.warn(s"PDV matched failed: ${pdvData.validationStatus}")
+        auditService.findYourNinoPDVMatchFailed(pdvData, origin)
+        redirectToInvalidDataNINOHelpController(mode, sessionWithNINO)
+      case "success" =>
+        handleSuccessValidationStatus(idDataError, origin, pdvData)
+      case _ =>
+        checkDetailsMatchingFailedWithUnknownIssue(mode)
+    }
+
+  private def handleSuccessValidationStatus(idDataError: IndividualDetailsError,
+                                            origin: Option[String],
+                                            pdvData: PDVResponseData
+                                           )(implicit hc: HeaderCarrier, request: DataRequest[AnyContent]): Result = {
+    auditService.findYourNinoPDVMatched(pdvData, origin, None)
 
-      auditService.findYourNinoIdDataError(pdvData, errorStatusCode, idDataError, origin)
-      logger.warn(s"Failed to retrieve Individual Details data: ${idDataError.errorMessage}")
-      Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
+    val errorStatusCode: Option[String] = idDataError match {
+      case conError: ConnectorError => Some(conError.statusCode.toString)
+      case _ => None
     }
+
+    auditService.findYourNinoIdDataError(pdvData, errorStatusCode, idDataError, origin)
+    logger.warn(s"Failed to retrieve Individual Details data: ${idDataError.errorMessage}")
+    // TODO review FailedDependency or use other status here???
+    FailedDependency(errorHandler.standardErrorTemplate())
```

Comment:
I think we need pagers if we get errors from API1694? We need to know if something has gone wrong. The important thing is to log the PDV success if PDV was successful but API1694 fails.